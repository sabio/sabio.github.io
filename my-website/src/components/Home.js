import React from 'react'
import ReactDOM from 'react-dom'

class Home extends React.Component{

  render(){
    return(
      <div className="componentContainer componentContainer-home">
      <div className="typewriter">
        <h1>Welcome to my humble website!</h1> 
        <br />
        <h2>Go ahead and check it out!</h2> 
      </div>
      </div>
    )
  }
}


export default Home
