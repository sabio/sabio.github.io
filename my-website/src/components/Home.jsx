import React from 'react'


const Home = () => (
  <div className="componentContainer componentContainer-home">
    <div className="typewriter">
      <h1>¡Bienvenid@s a mi website!</h1>
      <br />
      <h2>¡Pásele con confianza!</h2>
    </div>
  </div>
)


export default Home
