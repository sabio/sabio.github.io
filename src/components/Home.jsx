import React from 'react'
import Flecha from './Flecha'

const Home = () => (
  <div className="componentContainer componentContainer-home">
    <div className="typewriter">
      <h1>¡Bienvenid@s a mi website!</h1>
      <h2>¡Pásele con confianza!</h2>
    </div>
    <br />
    <br />
    <div className="divFlechas">    
        <Flecha direccion='der' ruta='/quiensoy' />
      </div>
  </div>
)


export default Home
