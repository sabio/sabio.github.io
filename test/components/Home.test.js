import React from 'react';
import Home from '../../src/components/Home';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { MemoryRouter } from 'react-router';
import { Link } from 'react-router-dom';

Enzyme.configure({ adapter: new Adapter() })

test('Rendering Home component', () => {
  const wrapper = mount(
    <MemoryRouter>
      <Home />
    </MemoryRouter>
  );
  expect(wrapper).toMatchSnapshot();

  expect(wrapper.find('h1')).toHaveLength(1);
  expect(wrapper.find('h2')).toHaveLength(1);
  expect(wrapper.find('Link')).toHaveLength(1);

  expect(wrapper.find('h1').text()).toEqual("¡Bienvenid@s a mi website!")
  expect(wrapper.find('h2').text()).toEqual("¡Pásele con confianza!")

  expect(wrapper.find(Link)).toHaveLength(1);
  expect(wrapper.find('Link').find('img')).toHaveLength(1);
  expect(wrapper.find('Link img')).toHaveLength(1);

  expect(wrapper.find('Link img').prop("alt")).toEqual("arrow");
  expect(wrapper.find('Link img').get(0).props).toHaveProperty("alt","arrow");
});