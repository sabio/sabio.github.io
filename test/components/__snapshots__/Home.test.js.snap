// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering Home component 1`] = `
<MemoryRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "met0xu",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "met0xu",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Home>
      <div
        className="componentContainer componentContainer-home"
      >
        <div
          className="typewriter"
        >
          <h1>
            ¡Bienvenid@s a mi website!
          </h1>
          <h2>
            ¡Pásele con confianza!
          </h2>
        </div>
        <br />
        <br />
        <div
          className="divFlechas"
        >
          <Flecha
            direccion="der"
            ruta="/quiensoy"
          >
            <Link
              to="/quiensoy"
            >
              <a
                href="/quiensoy"
                onClick={[Function]}
              >
                <img
                  alt="arrow"
                  className="flecha"
                  src={Object {}}
                />
              </a>
            </Link>
          </Flecha>
        </div>
      </div>
    </Home>
  </Router>
</MemoryRouter>
`;
